packages/build_package.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+set -e
+set +x
+
+if [[ "$(uname)" != "Darwin" && "$(uname)" != "Linux" ]]; then
+  echo "ERROR: can't run on $(uname). This script is designed to work on Mac or Linux only"
+  exit 1
+fi
+
+
+# 1. Read arguments
+
+if [[ $1 == "--help" ]]; then
+  echo "usage: $(basename $0) <package-name> <output-path> [--no-cleanup] [--no-build]"
+  echo
+  echo "Creates a .tgz of the package and saves it at the given output path"
+  echo
+  echo "  --no-cleanup    skip cleaning up generated files from package directory"
+  echo "  --no-build      skip building root package. DANGER: make sure to build root package before running this script."
+  echo
+  echo "Example:"
+  echo "  $(basename $0) playwright $(pwd -P)/playwright.tgz"
+  exit 1
+fi
+
+if [[ $# < 1 ]]; then
+  echo "Please specify package name, e.g. 'playwright' or 'playwright-chromium'."
+  echo "Try running $(basename $0) --help"
+  exit 1
+fi
+PACKAGE_NAME="$1"
+
+if [[ $# < 2 ]]; then
+  echo "Please specify output path"
+  echo "Try running $(basename $0) --help"
+  exit 1
+fi
+OUTPUT_PATH=$(node -e "console.log(require('path').resolve('$2'))")
+
+NO_CLEANUP=""
+if [[ "$3" == "--no-cleanup" || "$4" == "--no-cleanup" ]]; then
+  NO_CLEANUP="true"
+fi
+
+NO_BUILD=""
+if [[ "$3" == "--no-build" || "$4" == "--no-build" ]]; then
+  NO_BUILD="true"
+fi
+
+
+# 2. Change directory to the script directory and setup cleanup hooks.
+
+CLEANUP_PATHS=()
+function do_cleanup {
+  if [[ ! -z "${NO_CLEANUP}" ]]; then
+    return
+  fi
+  for CLEANUP_PATH in "${CLEANUP_PATHS[@]}"; do
+    rm -rf "${CLEANUP_PATH}"
+  done
+}
+trap "do_cleanup; cd $(pwd -P)" EXIT
+cd "$(dirname $0)"
+
+
+# 3. Figure out package parameters.
+PACKAGE_PATH="$(pwd -P)/${PACKAGE_NAME}"
+PACKAGE_DESCRIPTION=""
+PACKAGE_BROWSERS="[]"
+if [[ "${PACKAGE_NAME}" == "playwright" ]]; then
+  PACKAGE_DESCRIPTION="A high-level API to automate web browsers"
+  PACKAGE_BROWSERS="['chromium', 'firefox', 'webkit']"
+
+  # For Playwright, we need to copy README.md
+  README_PATH="${PACKAGE_PATH}/README.md"
+  CLEANUP_PATHS+=("${README_PATH}")
+  cp ../README.md "${README_PATH}"
+elif [[ "${PACKAGE_NAME}" == "playwright-core" ]]; then
+  PACKAGE_DESCRIPTION="A high-level API to automate web browsers"
+  PACKAGE_BROWSERS="[]"
+elif [[ "${PACKAGE_NAME}" == "playwright-webkit" ]]; then
+  PACKAGE_DESCRIPTION="A high-level API to automate Webkit"
+  PACKAGE_BROWSERS="['webkit']"
+elif [[ "${PACKAGE_NAME}" == "playwright-firefox" ]]; then
+  PACKAGE_DESCRIPTION="A high-level API to automate Firefox"
+  PACKAGE_BROWSERS="['firefox']"
+elif [[ "${PACKAGE_NAME}" == "playwright-chromium" ]]; then
+  PACKAGE_DESCRIPTION="A high-level API to automate Chromium"
+  PACKAGE_BROWSERS="['chromium']"
+else
+  echo "ERROR: unknown package '${PACKAGE_NAME}'"
+  exit 1
+fi
+
+
+# 4. Build root package unless --no-build was provided.
+
+if [[ -z "${NO_BUILD}" ]]; then
+  npm run build
```

Comment:
1. types are missing
2. .npmignore is missing
3. do not `npm run clean` and do not `npm run build`